CMakeLists.txt

```diff
@@ -32,6 +32,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake")
 set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 set(CMAKE_ENABLE_EXPORTS TRUE)
+option(CMAKE_INTERPROCEDURAL_OPTIMIZATION "Perform link time optimization" OFF)
```

Comment:
I thought the intention is to default this to ON, same as in Qpid Proton.

Otherwise, users can just set `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON` and there is no need for any additional CMakeLists code to support this thing.

Reply:
FWIW Proton does not make this an option - the intent is just to rely on the underlying cmake support and to remove the specific option.

Reply:
Proton does seem to have an option called `ENABLE_LINKTIME_OPTIMIZATION`
`option(ENABLE_LINKTIME_OPTIMIZATION "Perform link time optimization" ${DEFAULT_LINKTIME_OPTIMIZATION})`

https://github.com/apache/qpid-proton/blob/main/CMakeLists.txt#L149

Reply:
As I said above that is for backwards compatibility which you don't need as you have no previous setting for this.

by 'make this an option' I mean literally use the option() command in cmake - you can just use a command line setting to turn this on unless you want to default this to on in which case you will need to use some way of setting it.

Reply:
> I thought the intention is to default this to ON, same as in Qpid Proton.
> 
> Otherwise, users can just set `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON` and there is no need for any additional CMakeLists code to support this thing.

You got your wish.